src/IO/ADIOS/ParallelADIOS1IOHandler.cpp

```diff
@@ -36,74 +37,59 @@ namespace openPMD
 #       define ASSERT(CONDITION, TEXT) do{ (void)sizeof(CONDITION); } while( 0 )
 #   endif
 
-ParallelADIOS1IOHandler::ParallelADIOS1IOHandler(std::string const& path,
-                                                 AccessType at,
-                                                 MPI_Comm comm)
-        : AbstractIOHandler(path, at, comm),
-          m_impl{new ParallelADIOS1IOHandlerImpl(this, comm)}
-{
-    m_impl->init();
-}
-
-ParallelADIOS1IOHandler::~ParallelADIOS1IOHandler()
-{ }
-
-std::future< void >
-ParallelADIOS1IOHandler::flush()
-{
-    return m_impl->flush();
-}
-
-void
-ParallelADIOS1IOHandler::enqueue(IOTask const& i)
+ParallelADIOS1IOHandlerImpl::ParallelADIOS1IOHandlerImpl(AbstractIOHandler* handler,
+                                                         MPI_Comm comm)
+        : ADIOS1IOHandlerImpl{handler},
+          m_mpiInfo{MPI_INFO_NULL}
 {
-    switch( i.operation )
-    {
-        case Operation::CREATE_FILE:
-        case Operation::CREATE_PATH:
-        case Operation::CREATE_DATASET:
-        case Operation::OPEN_FILE:
-        case Operation::WRITE_ATT:
-            m_setup.push(i);
-            return;
-        default:
-            m_work.push(i);
-            return;
-    }
+    int status = MPI_SUCCESS;
+    status = MPI_Comm_dup(comm, &m_mpiComm);
+    ASSERT(status == MPI_SUCCESS, "Internal error: Failed to duplicate MPI communicator");
 }
 
-std::string
-getEnvNum(std::string const& key, std::string const& defaultValue)
+ParallelADIOS1IOHandlerImpl::~ParallelADIOS1IOHandlerImpl()
 {
-    char const* env = std::getenv(key.c_str());
-    if( env != nullptr )
+    /* create all files where ADIOS file creation has been deferred, but this has never been triggered
```

Comment:
the comment is not very clear: what is "this" what has never been triggered and what is "this" that happens in the line below?

Reply:
True, the sentence really is a bit vague.
First "this" means actual file creation. The operation (file creation) has been deferred, but because no IO op that triggers execution of the deferred operation (file creation) has been scheduled across the whole Series, we have to do it manually.
Seconds "this" stands for the situation described above (i.e. the scenarion in which no appropiate IO op has been scheduled).